testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/AnotherSseResource.java

```diff
@@ -25,24 +25,33 @@
    @Path("/subscribe")
    @Produces(MediaType.SERVER_SENT_EVENTS)
    public void subscribe(@Context SseEventSink sink) throws IOException {
-	   if (sink == null) {
-           throw new IllegalStateException("No client connected.");
-       }
-       //subscribe
-	   if (sseBroadcaster == null) {
-		   sseBroadcaster = sse.newBroadcaster();
-	   }
-	   sseBroadcaster.register(sink);	  
+      synchronized (outputLock)
+      {
+         if (sink == null)
+         {
+            throw new IllegalStateException("No client connected.");
+         }
+      }
+      eventSink = sink;
```

Comment:
So, the line here above (eventSink = sink;) is really a bug fix, as it stores a reference to the sink in the resource, allowing the close() method to actually do something. Nice catch.